### DISCUSSION TITLE
Возможность использования вычислений между особенностями трека (features)

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#87] [Помощь с алгоритмом] [answered]
Подсмотрел на OrganizeYourMusic (plamere) очень полезный параметр трека - "anger". Он удивительно точно описывает степень "злости/раздражительности" трека. Увязан этот параметр с двумя известными нам особенностями "Energy" и "Valence" таким образом:
`track.feats.anger = audio_feature.energy * (1 - audio_feature.valence);`
Существует ли возможность каким-либо образом использовать этот параметр в функции rangeTracks?

P.S.
Вот ещё небезынтересные зависимости (скопировал у plamere):
`track.feats.happiness = audio_feature.energy * audio_feature.valence;`
`track.feats.sadness=  (1 - audio_feature.energy) * (1 - audio_feature.valence);`

**Comment by Chimildic** [comment, marked as answer]
Тестируем в версии 1.4.8

**Reply by Bedmit** [reply]
Что-то смутило - "Нужно обновить файл config". Это как?

**Reply by Chimildic** [reply]
Появилось два параметра. Нужно их добавить и вызвать функцию `setProperties`.
Без обновления файла тоже будет работать со значениями по умолчанию. Просто для удобства.

**Reply by Bedmit** [reply]
> Нужно их добавить

Так правильно?
userProperties.setProperty('COUNT_RECENT_TRACKS', '30000');

И что писать в строке LOCALE? В описании не указано.

**Reply by Chimildic** [reply]
Там где основной код `library`, есть и файл `config`

**Reply by Bedmit** [reply]
Я понимаю, где писать. Я не понимаю, что.
Если в 
`userProperties.setProperty('COUNT_RECENT_TRACKS', '30000'); `
еще понятно, что можно указать число от 20000 до 40000, то что писать в LOCALE ?

**Reply by Chimildic** [reply]
Просто скопировать что есть. По умолчанию `RU`

**Reply by Bedmit** [reply]
Все равно не пойму. Откуда скопировать? (Реально ощущаю себя каким-то идиотом)

У меня в config
```
function setProperties(){
    let userProperties = PropertiesService.getUserProperties();
    userProperties.setProperty('CLIENT_ID', '123');
    userProperties.setProperty('CLIENT_SECRET', '123');
    userProperties.setProperty('LASTFM_API_KEY', '123');
    userProperties.setProperty('ON_SPOTIFY_RECENT_TRACKS', 'true');
    userProperties.setProperty('ON_LASTFM_RECENT_TRACKS', 'true');
    userProperties.setProperty('LASTFM_RANGE_RECENT_TRACKS', '30');
    userProperties.setProperty('LASTFM_LOGIN', '123');
}

function logProperties(){
    console.log(PropertiesService.getUserProperties().getProperties());
}
```
Я понял, что нужно вставить еще две строки. Первая - `userProperties.setProperty('COUNT_RECENT_TRACKS', '30000');`
А какая вторая? - `userProperties.setProperty('LOCALE', 'вот тут что писать?');`  -   `RU`?

**Reply by Chimildic** [reply]
Да, `RU`, [линк](https://github.com/Chimildic/goofy/blob/main/config.js)

**Reply by Bedmit** [reply]
Черт! Только теперь прозрел. Просто оказалось, что фазу "Там где основной код library" я понимал, как свою библиотеку. И соответственно как свой конфиг.
А вообще пытался сам дописать код, ориентируясь на описание параметров из инструкции. ИМХО в описание параметра  `LOCALE` стоит добавить что в строку нужно вставить `RU`
